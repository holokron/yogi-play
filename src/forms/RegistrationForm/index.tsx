import * as React from 'react'
import { Button, FormGroup, Input, UncontrolledAlert } from 'reactstrap'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export interface Props {
    onSubmit(email: string, password: string): Promise<any>
}

export interface State {
    email: any
    password: any
    isLoading: boolean
    error: boolean
}

export default class RegistrationForm extends React.PureComponent<Props, State> {
    state: State = {
        email: '',
        password: '',
        isLoading: false,
        error: false,
    }

    constructor(props: Props) {
        super(props)

        this.handleChangeEmail = this.handleChangeEmail.bind(this)
        this.handleChangePassword = this.handleChangePassword.bind(this)
        this.handleSubmit = this.handleSubmit.bind(this)
    }

    handleChangeEmail(event: React.ChangeEvent<HTMLInputElement>): void {
        this.setState({
            email: event.target.value,
            error: false,
        })
    }

    handleChangePassword(event: React.ChangeEvent<HTMLInputElement>): void {
        this.setState({
            password: event.target.value,
            error: false,
        })
    }

    async handleSubmit(event: React.FormEvent<HTMLFormElement>): Promise<void> {
        event.preventDefault()

        this.setState({
            isLoading: true,
            error: false,
        })

        const {
            email,
            password,
        } = this.state

        try {
            await this.props.onSubmit(email, password)
        } catch (error) {
            this.setState({
                isLoading: false,
                error: true,
            })
        }
    }

    renderErrors() {
        if (!this.state.error) {
            return null
        }

        return (
            <UncontrolledAlert color="danger">
                Nieprawidłowy email i/lub hasło
            </UncontrolledAlert>
        )
    }

    render() {
        return (
            <form onSubmit={this.handleSubmit} noValidate>
                {this.renderErrors()}
                <FormGroup>
                    <label>Email</label>
                    <Input
                        type="email" 
                        name="email" 
                        placeholder="Email"
                        onChange={this.handleChangeEmail}
                        value={this.state.email}
                    />
                </FormGroup>
                <FormGroup>
                    <label>Hasło</label>
                    <Input
                        type="password"
                        name="password"
                        placeholder="Hasło"
                        onChange={this.handleChangePassword}
                        value={this.state.password}
                    />
                </FormGroup>
                <Button color="primary" type="submit">
                    {this.state.isLoading 
                        ? <React.Fragment>
                            <FontAwesomeIcon icon="spinner" fixedWidth pulse /> Przetwarzanie...
                            </React.Fragment>
                        : 'Zarejestruj'
                    }
                </Button>
            </form>
        )
    }
}