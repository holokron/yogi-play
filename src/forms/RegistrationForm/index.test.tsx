import * as React from 'react'
import * as renderer from 'react-test-renderer'
import RegistrationForm from '.'

describe('@forms/RegistrationForm', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <RegistrationForm 
                onSubmit={(emai: string, password: string): Promise<any> => new Promise(() => {})}
            />
        ).toJSON()
        
        expect(tree).toMatchSnapshot()
    })
})