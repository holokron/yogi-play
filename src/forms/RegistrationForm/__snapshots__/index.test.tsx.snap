// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@forms/RegistrationForm renders correctly 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="position-relative form-group"
  >
    <label>
      Email
    </label>
    <input
      className="form-control"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
      value=""
    />
  </div>
  <div
    className="position-relative form-group"
  >
    <label>
      Hasło
    </label>
    <input
      className="form-control"
      name="password"
      onChange={[Function]}
      placeholder="Hasło"
      type="password"
      value=""
    />
  </div>
  <button
    aria-label={null}
    className="btn btn-primary"
    onClick={[Function]}
    type="submit"
  >
    Zarejestruj
  </button>
</form>
`;
